components/ruler/index.vue

```diff
@@ -0,0 +1,357 @@
+<template>
+  <div
+    class="md-ruler"
+    @touchstart="$_startDrag"
+    @touchend="$_stopDrag"
+  >
+    <canvas
+      class="md-ruler-canvas"
+      ref="canvas"
+    ></canvas>
+    <div class="md-ruler-cursor"></div>
+    <div class="md-ruler-arrow"></div>
+  </div>
+</template>
+
+<script>
import Scroller from '../_util/scroller'
+
+export default {
+  name: 'md-ruler',
+
+  components: {},
+
+  props: {
+    value: {
+      type: Number,
+      default: 0,
+    },
+    scope: {
+      type: Array,
+      default: () => [0, 100],
+    },
+    step: {
+      type: Number,
+      default: 10,
+    },
+    unit: {
+      type: Number,
+      default: 1,
+    },
+    min: {
+      type: Number,
+      default: 0,
+    },
+    max: {
+      type: Number,
+      default: 100,
+    },
+  },
+
+  data() {
+    return {
+      ctx: null, // canvas
```

Comment:
data is only for ui reactive, remove `ctx` here and assign `ctx` under `this` context directly.

Reply:
I don't think so data is only for ui reactive.Put `ctx` here for other developer understand where the `ctx` is defined.